HDDS-1689. Implement S3 Create Bucket request to use Cache and DoubleBuffer.

bharatviswa504 opened the pull request.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 79 | Maven dependency ordering for branch |
| +1 | mvninstall | 496 | trunk passed |
| +1 | compile | 242 | trunk passed |
| +1 | checkstyle | 61 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 778 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 147 | trunk passed |
| 0 | spotbugs | 315 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 497 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 74 | Maven dependency ordering for patch |
| +1 | mvninstall | 452 | the patch passed |
| +1 | compile | 351 | the patch passed |
| +1 | cc | 351 | the patch passed |
| +1 | javac | 351 | the patch passed |
| -0 | checkstyle | 35 | hadoop-ozone: The patch generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 625 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 162 | the patch passed |
| +1 | findbugs | 600 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 282 | hadoop-hdds in the patch passed. |
| -1 | unit | 2039 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 48 | The patch does not generate ASF License warnings. |
| | | 7252 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestContainerStateMachineFailures |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 22528066b727 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 4a70a0d |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/1/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/1/testReport/ |
| Max. process+thread count | 5307 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 33 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 52 | Maven dependency ordering for branch |
| +1 | mvninstall | 507 | trunk passed |
| +1 | compile | 253 | trunk passed |
| +1 | checkstyle | 62 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 843 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 144 | trunk passed |
| 0 | spotbugs | 320 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 511 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| +1 | mvninstall | 446 | the patch passed |
| +1 | compile | 253 | the patch passed |
| +1 | cc | 253 | the patch passed |
| +1 | javac | 253 | the patch passed |
| -0 | checkstyle | 32 | hadoop-ozone: The patch generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 643 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 158 | the patch passed |
| +1 | findbugs | 574 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 244 | hadoop-hdds in the patch failed. |
| -1 | unit | 2532 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 47 | The patch does not generate ASF License warnings. |
| | | 7529 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.web.client.TestKeys |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestReadRetries |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux e5f2f70c083a 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 4a70a0d |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/3/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/3/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/3/testReport/ |
| Max. process+thread count | 5329 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 88 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 77 | Maven dependency ordering for branch |
| +1 | mvninstall | 550 | trunk passed |
| +1 | compile | 268 | trunk passed |
| +1 | checkstyle | 74 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 949 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 168 | trunk passed |
| 0 | spotbugs | 374 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 604 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 33 | Maven dependency ordering for patch |
| +1 | mvninstall | 508 | the patch passed |
| +1 | compile | 271 | the patch passed |
| +1 | cc | 271 | the patch passed |
| +1 | javac | 271 | the patch passed |
| -0 | checkstyle | 39 | hadoop-ozone: The patch generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 751 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 193 | the patch passed |
| +1 | findbugs | 678 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 358 | hadoop-hdds in the patch passed. |
| -1 | unit | 2912 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 53 | The patch does not generate ASF License warnings. |
| | | 8777 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestBlockOutputStream |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestFailureHandlingByClient |
|   | hadoop.ozone.client.rpc.TestKeyInputStream |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux d73f0f46d68f 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 4a70a0d |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/2/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/2/testReport/ |
| Max. process+thread count | 4875 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 62 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 113 | Maven dependency ordering for branch |
| +1 | mvninstall | 649 | trunk passed |
| +1 | compile | 271 | trunk passed |
| +1 | checkstyle | 89 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 927 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 155 | trunk passed |
| 0 | spotbugs | 317 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 518 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 33 | Maven dependency ordering for patch |
| +1 | mvninstall | 531 | the patch passed |
| +1 | compile | 283 | the patch passed |
| +1 | cc | 283 | the patch passed |
| +1 | javac | 283 | the patch passed |
| +1 | checkstyle | 77 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 638 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 155 | the patch passed |
| +1 | findbugs | 564 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 263 | hadoop-hdds in the patch failed. |
| -1 | unit | 2486 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 50 | The patch does not generate ASF License warnings. |
| | | 8023 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.ozoneimpl.TestOzoneContainer |
|   | hadoop.ozone.client.rpc.TestBlockOutputStream |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.TestMiniOzoneCluster |
|   | hadoop.ozone.client.rpc.TestReadRetries |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 3621b3acb006 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 64d4abf |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/4/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/4/testReport/ |
| Max. process+thread count | 4983 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 35 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for branch |
| +1 | mvninstall | 506 | trunk passed |
| +1 | compile | 243 | trunk passed |
| +1 | checkstyle | 60 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 797 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 152 | trunk passed |
| 0 | spotbugs | 309 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 498 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 31 | Maven dependency ordering for patch |
| +1 | mvninstall | 436 | the patch passed |
| +1 | compile | 255 | the patch passed |
| +1 | cc | 255 | the patch passed |
| +1 | javac | 255 | the patch passed |
| +1 | checkstyle | 72 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 629 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 163 | the patch passed |
| +1 | findbugs | 573 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 282 | hadoop-hdds in the patch passed. |
| -1 | unit | 2578 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 40 | The patch does not generate ASF License warnings. |
| | | 7543 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestKeyInputStream |
|   | hadoop.ozone.client.rpc.TestHybridPipelineOnDatanode |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.TestSecureOzoneCluster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 83d9967a1ed6 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1411513 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/5/testReport/ |
| Max. process+thread count | 4132 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeCreateRequest.java:
```diff
@@ -145,28 +136,24 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
             OMException.ResultCodes.VOLUME_ALREADY_EXISTS);
       }
 
-      volumeList = omMetadataManager.getUserTable().get(dbUserKey);
```
Was this line of code lost?

bharatviswa504 replied:
Thanks for catching. It is a coding mistake. I have fixed it, and also added a test case for it.

arp7 commented on hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/bucket/package-info.java:
```diff
@@ -0,0 +1,23 @@
+/**
+ * Package contains test classes for s3 bucket requests.
+ */
+package org.apache.hadoop.ozone.om.request.s3.bucket;
```
Add newline at end of file.

bharatviswa504 replied:
Done

arp7 replied:
Sorry I still don't see a newline. It's a minor point. See https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

This is an old habit from using a C++ compiler that would throw errors on files not ending in newlines. 🙂

bharatviswa504 replied:
Added the new line, in my IDE is see it, not sure GitHub UI is not showing that.

![Screen Shot 2019-07-17 at 4 42 01 PM](https://user-images.githubusercontent.com/8586345/61419169-dce23200-a8b1-11e9-8661-ee7813ded2e3.png)

bharatviswa504 replied:
![Screen Shot 2019-07-17 at 4 42 01 PM](https://user-images.githubusercontent.com/8586345/61419182-f5eae300-a8b1-11e9-8863-5a7f8c874f03.png)

bharatviswa504 replied:
![Screen Shot 2019-07-17 at 4 43 02 PM](https://user-images.githubusercontent.com/8586345/61419202-03a06880-a8b2-11e9-9ba9-11428b9d96bf.png)

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/bucket/S3BucketCreateResponse.java:
```diff
@@ -0,0 +1,72 @@
+          s3Mapping);
+    }
+  }
+}
```
Add newline please. I missed this in general, it's good to have files end in newlines.

Perhaps not true these days but historically some tools used to choke on text files that don't end in newline.

bharatviswa504 replied:
Done

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/bucket/S3BucketCreateResponse.java:
```diff
@@ -0,0 +1,72 @@
+        omVolumeCreateResponse.addToDBBatch(omMetadataManager, batchOperation);
+      }
+
+      if (omBucketCreateResponse != null) {
```
If status is okay then omBucketCreateResponse must be non-null right?

bharatviswa504 replied:
Yes. But this check is added, because we have fields in constructor annotated as nullable, so find bugs will catch us, to avoid that added this check.

arp7 replied:
Instead of that you can add a Preconditions.checkState.

bharatviswa504 replied:
Done.

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/bucket/S3BucketCreateRequest.java:
```diff
@@ -0,0 +1,350 @@
+    // For now only checked the length.
+    int bucketLength = s3CreateBucketRequest.getS3Bucketname().length();
+    if (bucketLength < 3 || bucketLength >= 64) {
+      throw new OMException("S3BucketName must be at least 3 and not more " +
```
Can we define 3 and 64 as constants?

bharatviswa504 replied:
Done

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/bucket/S3BucketCreateRequest.java:
```diff
@@ -0,0 +1,350 @@
+  }
+
+  @Override
+  @SuppressWarnings("methodlength")
```
Not feasible to reduce the method length? 🙂

bharatviswa504 replied:
Moved create bucket to a new method.

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/bucket/S3BucketCreateRequest.java:
```diff
@@ -0,0 +1,350 @@
+
+    S3CreateBucketRequest s3CreateBucketRequest =
+        getOmRequest().getCreateS3BucketRequest();
+    Preconditions.checkNotNull(s3CreateBucketRequest);
```
Don't check for null ref just before dereferencing the null ref. It is redundant.

bharatviswa504 replied:
Done. The actual intention here to do is, whether S3BucketCreateRequest is called with OMRequest which has S3BucketCreateRequest or not. I removed the check.

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/bucket/S3BucketCreateRequest.java:
```diff
@@ -0,0 +1,350 @@
+        }
+      }
+
+      // Now finally add it to s3 table cache.
```
IIUC the same bucket is cached twice, once in bucket table and once in S3 table.

bharatviswa504 replied:
In S3Table we cache S3Bucketname -> OzonevolumeName/OzoneBucketName. In S3 world buckets are unique, to acheive this, we have this S3Table.
In Bucket Table we cache BucketName to BucketInfo, not this mapping.

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/bucket/S3BucketCreateRequest.java:
```diff
@@ -0,0 +1,350 @@
+   * @return s3Mapping
+   */
+  @VisibleForTesting
+  public static String formatS3MappingName(String volumeName,
```
Dumb q. What does this function do?

bharatviswa504 replied:
This generates an s3Mapping for volumeName and bucketName which we store in s3Table.

bharatviswa504 commented:
Thank You @arp7 for the review.
I have addressed the review comments.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 59 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 6 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 71 | Maven dependency ordering for branch |
| +1 | mvninstall | 550 | trunk passed |
| +1 | compile | 261 | trunk passed |
| +1 | checkstyle | 70 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 848 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 156 | trunk passed |
| 0 | spotbugs | 335 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 529 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 32 | Maven dependency ordering for patch |
| +1 | mvninstall | 448 | the patch passed |
| +1 | compile | 248 | the patch passed |
| +1 | cc | 248 | the patch passed |
| +1 | javac | 248 | the patch passed |
| +1 | checkstyle | 66 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 618 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 84 | hadoop-ozone generated 2 new + 13 unchanged - 0 fixed = 15 total (was 13) |
| -1 | findbugs | 100 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 331 | hadoop-hdds in the patch failed. |
| -1 | unit | 101 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 5019 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestContainerPlacementFactory |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.8 Server=18.09.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 3310cd41fd9c 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 303a7f8 |
| Default Java | 1.8.0_212 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/6/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/6/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/6/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/6/testReport/ |
| Max. process+thread count | 487 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 37 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 6 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 64 | Maven dependency ordering for branch |
| +1 | mvninstall | 511 | trunk passed |
| +1 | compile | 265 | trunk passed |
| +1 | checkstyle | 76 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 866 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 156 | trunk passed |
| 0 | spotbugs | 326 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 523 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 32 | Maven dependency ordering for patch |
| +1 | mvninstall | 442 | the patch passed |
| +1 | compile | 253 | the patch passed |
| +1 | cc | 253 | the patch passed |
| +1 | javac | 253 | the patch passed |
| +1 | checkstyle | 75 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 648 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 152 | the patch passed |
| +1 | findbugs | 531 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 272 | hadoop-hdds in the patch failed. |
| -1 | unit | 1397 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 37 | The patch does not generate ASF License warnings. |
| | | 6487 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestContainerPlacementFactory |
|   | hadoop.ozone.om.TestOzoneManager |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.container.ozoneimpl.TestSecureOzoneContainer |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.8 Server=18.09.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 081a3ea80694 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 303a7f8 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/7/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/7/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/7/testReport/ |
| Max. process+thread count | 5095 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/7/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

bharatviswa504 commented:
Thank You @arp7 for the review.
I have addressed the review comments. And regrading newline on Github UI it is not showing up, when you see in IDE it is showing up.

bharatviswa504 commented:
/retest

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 39 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 6 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 25 | Maven dependency ordering for branch |
| +1 | mvninstall | 510 | trunk passed |
| +1 | compile | 274 | trunk passed |
| +1 | checkstyle | 80 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 871 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 163 | trunk passed |
| 0 | spotbugs | 318 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 509 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 74 | Maven dependency ordering for patch |
| +1 | mvninstall | 451 | the patch passed |
| +1 | compile | 276 | the patch passed |
| +1 | cc | 276 | the patch passed |
| +1 | javac | 276 | the patch passed |
| +1 | checkstyle | 86 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 667 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 166 | the patch passed |
| +1 | findbugs | 528 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 295 | hadoop-hdds in the patch failed. |
| -1 | unit | 1718 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 6969 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestContainerPlacementFactory |
|   | hadoop.ozone.container.server.TestSecureContainerServer |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.container.ozoneimpl.TestSecureOzoneContainer |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.8 Server=18.09.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux ef49d891ac3b 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 5e6cc6f |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/8/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/8/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/8/testReport/ |
| Max. process+thread count | 5388 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/8/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 58 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 6 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 23 | Maven dependency ordering for branch |
| +1 | mvninstall | 529 | trunk passed |
| +1 | compile | 337 | trunk passed |
| +1 | checkstyle | 61 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 785 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 149 | trunk passed |
| 0 | spotbugs | 303 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 498 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 28 | Maven dependency ordering for patch |
| +1 | mvninstall | 414 | the patch passed |
| +1 | compile | 244 | the patch passed |
| +1 | cc | 244 | the patch passed |
| +1 | javac | 244 | the patch passed |
| +1 | checkstyle | 63 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 608 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 145 | the patch passed |
| +1 | findbugs | 527 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 342 | hadoop-hdds in the patch failed. |
| -1 | unit | 227 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 5228 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestContainerPlacementFactory |
|   | hadoop.ozone.om.ratis.TestOzoneManagerDoubleBufferWithOMResponse |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.8 Server=18.09.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux fcfc44856afe 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 73e6ffc |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/9/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/9/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/9/testReport/ |
| Max. process+thread count | 486 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/9/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 86 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 6 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 83 | Maven dependency ordering for branch |
| +1 | mvninstall | 517 | trunk passed |
| +1 | compile | 269 | trunk passed |
| +1 | checkstyle | 71 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 901 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 163 | trunk passed |
| 0 | spotbugs | 352 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 564 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 33 | Maven dependency ordering for patch |
| +1 | mvninstall | 428 | the patch passed |
| +1 | compile | 278 | the patch passed |
| +1 | cc | 278 | the patch passed |
| +1 | javac | 278 | the patch passed |
| +1 | checkstyle | 152 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 740 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 155 | the patch passed |
| +1 | findbugs | 535 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 341 | hadoop-hdds in the patch failed. |
| -1 | unit | 2198 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 7736 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestContainerPlacementFactory |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestKeyInputStream |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.container.ozoneimpl.TestSecureOzoneContainer |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.container.server.TestSecureContainerServer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.8 Server=18.09.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/10/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1088 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 386c9253f42b 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 73e6ffc |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/10/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/10/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/10/testReport/ |
| Max. process+thread count | 5403 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1088/10/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

bharatviswa504 commented:
Test failure is not related to this patch.
Thank You @arp7 for the review.
I will commit this to the trunk.
